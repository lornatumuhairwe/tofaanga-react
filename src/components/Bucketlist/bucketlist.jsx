import React, {Component} from 'react';
import { Modal, OverlayTrigger, Tooltip, Button } from 'react-bootstrap';
import '../../styles/css/bootstrap.min.css';
import '../../styles/css/bucketlist.css';
import BucketlistTable from './bucketlistTable';
import LoginForm from '../Authentication/Login/Login';
import {baseUrl} from '../../constants';
import Search from '../Search/search';
import NotificationSystem from 'react-notification-system';
import Loader from '../loader'
import PaginationComp from '../Pagination/pagination';

export default class Bucketlist extends Component {
    constructor(props) {
        super(props);
        this.state = {
            bucketlists: '',
            showModal: false,
            bname: '',
            isLoggedIn: this.props.isLoggedIn,
            details: '',
            notificationSystem: null, loading: true
        };
        this.closeModal = this.closeModal.bind(this);
        this.openModal = this.openModal.bind(this);
        this.addBucketlist = this.addBucketlist.bind(this);
        this.handleAdd = this.handleAdd.bind(this);
        // this.getName = this.getName.bind(this);
        this.getBucketlists = this.getBucketlists.bind(this);
        this.LogOut = this.LogOut.bind(this);
    }

    componentDidMount() {
        this.setState({notificationSystem: this.refs.notificationSystem});
    }

    getBucketlists(url) {

        const getData = {
            method: 'GET',
            headers: {Authorization: localStorage.getItem('token')},
        };
        return fetch(url, getData)
            .then((response) => {
                if (response.status === 401) {
                    localStorage.removeItem('token');
                    this.setState({isLoggedIn: false});
                } else {
                    return response.json()
                }
            }).then(res => {
                    if (res['bucketlists']) {
                        this.setState({bucketlists: res['bucketlists'], details: res['details'], loading: false});
                    }
                    else{
                        this.setState({bucketlists: '' });
                    }
                }
            )
    }

    closeModal() {
        this.setState({showModal: false});
    }

    openModal() {
        this.setState({showModal: true});
    }

    getName = (field, event) => {
        const newState = {};
        newState[field] = event.target.value;
        this.setState(newState);
    };

    LogOut(event) {
        localStorage.removeItem('token');
        event.preventDefault();
        this.setState({isLoggedIn: false});
    }

    addBucketlist(url, data) {
        const postData = {
            method: 'POST',
            body: data,
            headers: {Authorization: localStorage.getItem('token')},
        };
        return fetch(url, postData)
            .then(response => response.json())
            .then((resjson) => {
                if (resjson.bucketlist) {
                    this.state.notificationSystem.addNotification({
                        message: 'Bucketlist added successfully',
                        level: 'success',
                    });
                    this.getBucketlists(`${baseUrl}/bucketlists/`);
                }
                else {
                    this.state.notificationSystem.addNotification({
                        message: resjson.message,
                        level: 'error',
                    });
                }
            });
    }

    handleAdd(event) {
        event.preventDefault();
        const BlData = new FormData();
        BlData.append('name', this.state.bname);
        this.addBucketlist(`${baseUrl}/bucketlists/`, BlData);
        this.setState({showModal: false});
        this.setState({bname: ''});
    }

    componentWillMount() {
        this.getBucketlists(`${baseUrl}/bucketlists/`);
    }

    render() {
        const addBucketlisttooltip = (
            <Tooltip id="tooltip">Add bucketlist.</Tooltip>
        );
        if (this.state.isLoggedIn) {
            return (<div>
                    <NotificationSystem ref="notificationSystem"/>
                    <nav className="navbar navbar-inverse">
                        <ul className="nav navbar-nav">
                            <li><a href="">TOFAANGA</a></li>
                        </ul>
                        <ul className="nav navbar-nav navbar-right">
                            <li id="logoutBtn"><a onClick={this.LogOut}>Logout</a></li>
                        </ul>
                        <Search getBucketlists={this.getBucketlists}/>
                    </nav>
                    <div className="container">
                    <OverlayTrigger placement="top" overlay={addBucketlisttooltip}>
                        <Button id="addBucketlist" className="btn btn-lg btn-primary btn-circle fixed" onClick={this.openModal}>
                            <i className="glyphicon glyphicon-plus"/>
                        </Button>
                    </OverlayTrigger>
                    {this.state.loading
                        ? <Loader/>
                        : null
                    }
                    <BucketlistTable
                        bucketlists={this.state.bucketlists}
                        token={localStorage.getItem('token')}
                        getBucketlists={this.getBucketlists}
                        details={this.state.details}
                    />
                    <PaginationComp details={this.state.details} getBucketlists={this.getBucketlists} />
                    <Modal
                        show={this.state.showModal}
                        onHide={this.closeModal}
                        {...this.props}
                        bsSize="small"
                        aria-labelledby="contained-modal-title-sm"
                    >
                        <Modal.Header closeButton>
                            <Modal.Title>Add Bucketlist</Modal.Title>
                        </Modal.Header>
                        <Modal.Body>
                            <form onSubmit={this.handleAdd}>
                                <div className="form-group">
                                    <label htmlFor="name">Name:</label>
                                    <input
                                        type="text"
                                        className="form-control"
                                        id="name"
                                        value={this.state.bname}
                                        onChange={this.getName.bind(this, 'bname')}
                                    />
                                </div>
                                <button type="submit" id="addBLBtn" className="btn btn-default">Add Bucketlist</button>
                            </form>
                        </Modal.Body>
                    </Modal>
                    </div>
                </div>
            );
        }

        return (<LoginForm displaySignup={this.props.displaySignup}/>);
    }
}
