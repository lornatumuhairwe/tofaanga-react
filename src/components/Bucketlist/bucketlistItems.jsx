import React, { Component } from 'react';
import { Modal, Panel, OverlayTrigger, Tooltip, Button } from 'react-bootstrap';
import '../../styles/css/bootstrap.min.css';
import '../../styles/css/bucketlist.css';
import { baseUrl } from "../../constants";
import NotificationSystem from 'react-notification-system';

export class BLIRow extends Component {
  constructor(props) {
    super(props);
    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
    this.handleUpdate = this.handleUpdate.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.deleteBucketlistItem = this.deleteBucketlistItem.bind(this);
    this.updateBucketlistItemsAction = this.updateBucketlistItemsAction.bind(this);
    this.deleteBucketlistItemsAction = this.deleteBucketlistItemsAction.bind(this);
    this.state = { showUpdate: false, title: '', deadline: '', status: '', notificationSystem: null };
  }

    componentDidMount(){
        this.setState({ notificationSystem: this.refs.notificationSystem });
    }

  deleteBucketlistItem(event) {
    event.preventDefault();
    const blitemID = this.refs.bid.value;
    const bID = this.props.bID;
      //eslint-disable-next-line
    const getDeleteConfirmation = confirm('Do you really want to delete this? ');
      if (getDeleteConfirmation === true) {
          this.deleteBucketlistItemsAction(`${baseUrl}/bucketlists/` + bID.toString() + '/items/' + blitemID.toString());
          this.props.getBucketlistItems(event);
      }
  }

  deleteBucketlistItemsAction(url){
      const putData = {
          method: 'DELETE',
          headers: { Authorization: localStorage.getItem('token') },
      };
      return fetch(url, putData)
          .then(response => response.json()).then((res) => {
              },
          );
  }

  openModal(event) {
    event.preventDefault();
    this.setState({ showUpdate: true });
  }

  closeModal(event) {
    event.preventDefault();
    this.setState({ showUpdate: false });
  }

  handleUpdate(event) {
    event.preventDefault();
    const blitemID = this.refs.bid.value;
    const bID = this.props.bID;
    const blItemData = new FormData();
    if (this.state.title === ''){
        const stateObject = this.state;
        stateObject.title = this.props.bucketListItem[0];
    }
    if (this.state.deadline === ''){
        const stateObject = this.state;
        stateObject.deadline = this.props.bucketListItem[1];
      }
    if (this.state.status === ''){
        const stateObject = this.state;
        stateObject.status = this.props.bucketListItem[2];
      }
    blItemData.append('title', this.state.title);
    blItemData.append('deadline', this.state.deadline);
    blItemData.append('status', this.state.status);
    this.props.getBucketlistItems(event);
    this.updateBucketlistItemsAction(`${baseUrl}/bucketlists/` + bID.toString() + '/items/' + blitemID.toString(), blItemData);
  }

  updateBucketlistItemsAction(url, data) {
    const putData = {
        method: 'PUT',
        body: data,
        headers: { Authorization: localStorage.getItem('token') },
    };
    return fetch(url, putData)
        .then(response => response.json()).then((res) => {
            if (res.message==="Bucketlist item updated successfully"){
                this.state.notificationSystem.addNotification({
                    message: res.message,
                    level: 'success'
                });
                this.setState({ showUpdate: false });
            } else {
                this.state.notificationSystem.addNotification({
                    message: res.message,
                    level: 'error'
                });
            }
            },
        );
}

  handleChange(field, event) {
    const newState = {};
    newState[field] = event.target.value;
    this.setState(newState);
  }

  render() {
      const editItemtooltip = (
          <Tooltip id="tooltip">Edit bucketlist item.</Tooltip>
      );
      const deleteItemtooltip = (
          <Tooltip id="tooltip">Delete bucketlist item.</Tooltip>
      );
    return (
      <tr>
        <td>{this.props.bucketListItem[0]}</td>
          <td>{this.props.bucketListItem[2]!=='Complete' ? this.props.bucketListItem[1] : "Congrats!!"}</td>
        <td>{this.props.bucketListItem[2]}</td>
        <td><div className="pull-right">
            <NotificationSystem ref="notificationSystem" />
          <form className="form-inline">
            <input
              type="hidden"
              name="name"
              value={this.props.id}
              ref="bid"
              required
            />
            <OverlayTrigger placement="top" overlay={editItemtooltip}>
            <Button type="submit" className="btn btn-xs btn-primary btn-circle " onClick={this.openModal}>
              <i className="glyphicon glyphicon-edit" /></Button>
            </OverlayTrigger>
            <OverlayTrigger placement="top" overlay={deleteItemtooltip}>
            <Button type="submit" className="btn btn-xs btn-danger btn-circle" onClick={this.deleteBucketlistItem}>
              <i className="glyphicon glyphicon-trash" /></Button>
            </OverlayTrigger>
            <Modal show={this.state.showUpdate} onHide={this.closeModal}
                   bsSize="small" aria-labelledby="contained-modal-title-sm" >
              <Modal.Header closeButton>
                <Modal.Title>Update item: {this.props.bucketListItem[0]}</Modal.Title>
              </Modal.Header>
              <Modal.Body>
                <form>
                  <div className="form-group">
                    <label htmlFor="title">Title:</label>
                    <input
                      type="text"
                      className="form-control"
                      id="name"
                      required
                      defaultValue={this.props.bucketListItem[0]}
                      onChange={this.handleChange.bind(this, 'title')}
                    />
                  </div>
                  <div className="form-group">
                    <label htmlFor="Deadline">Deadline:</label>
                    <input
                      type="date"
                      className="form-control"
                      id="deadline"
                      defaultValue={this.props.bucketListItem[1]}
                      onChange={this.handleChange.bind(this, 'deadline')}
                    />
                  </div>
                  <div className="form-group">
                    <label htmlFor="status">Status: </label>
                    <select name="status"
                            onChange={this.handleChange.bind(this, 'status')}
                            value={this.state.status}>
                      <option value={this.props.bucketListItem[2]}>{this.props.bucketListItem[2]}</option>
                      <option value="Incomplete">Incomplete</option>
                      <option value="Complete">Complete</option>
                    </select>
                  </div>
                  <button type="submit" className="btn btn-default" onClick={this.handleUpdate}>Update item</button>
                </form>
              </Modal.Body>
            </Modal>
          </form>
        </div>
        </td>
      </tr>
    );
  }
}

export default class BucketlistItems extends Component {

  render() {
    const rows = [];
    const blsi = this.props.items;
    for (const key in blsi) {
      if (blsi.hasOwnProperty(key)) {
        rows.push(<BLIRow bID={this.props.bID} id={key} key={key}
                          bucketListItem={blsi[key]} token={this.props.token}
                          getBucketlistItems={this.props.getBucketlistItems} />);
      }
    }
    return (<div className="panel">
          <Panel className="panelpanel" collapsible expanded={this.props.showItemPanel} >
            <table className="table table-responsive table-hover">
              <thead>
              <tr>
                <th>Item</th>
                <th>Deadline</th>
                <th>Status</th>
              </tr>
              </thead>
              <tbody>
              {rows}
              </tbody>
            </table>
        </Panel>
    </div>
    );
  }
}

