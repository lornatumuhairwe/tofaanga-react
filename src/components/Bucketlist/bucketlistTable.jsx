import React from 'react';
import '../../styles/css/bootstrap.min.css';
import '../../styles/css/bucketlist.css';
import BLRow from './bucketlistTableRows';

export default class BucketlistTable extends React.Component {
  render() {
    const rows = [];
    const bls = this.props.bucketlists;
    if (Object.keys(bls).length === 0) {
      rows.push(<li key={0} className="list-group-item col-md-12 item">
        You don't have any bucketlists! Add some and start tracking.</li>);
    } else {
      for (const key in bls) {
        if (bls.hasOwnProperty(key)) {
          rows.push(<BLRow
            key={key}
            id={key}
            bucketlist={bls[key]}
            token={this.props.token}
            getBucketlists={this.props.getBucketlists}
          />);
        }
      }
    }

    return (<div>
      <ul className="list-group col-md-12" id="bucketlists">
        {rows}
      </ul>
    </div>
    );
  }
}

