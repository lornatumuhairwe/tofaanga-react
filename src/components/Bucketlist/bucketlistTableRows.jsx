import React from 'react';
import { Modal, OverlayTrigger, Tooltip, Button } from 'react-bootstrap';
import '../../styles/css/bootstrap.min.css';
import '../../styles/css/bucketlist.css';
import BucketlistItems from './bucketlistItems';
import { baseUrl } from "../../constants";
import NotificationSystem from 'react-notification-system';


export default class BLRow extends React.Component {
    constructor(props) {
        super(props);
        this.deleteBucketlist = this.deleteBucketlist.bind(this);
        this.editBucketlist = this.editBucketlist.bind(this);
        this.deleteAction = this.deleteAction.bind(this);
        this.updateAction = this.updateAction.bind(this);
        this.openModal = this.openModal.bind(this);
        this.openAdd = this.openAdd.bind(this);
        this.closeModal = this.closeModal.bind(this);
        this.getBucketlistItems = this.getBucketlistItems.bind(this);
        this.getBucketlistItemsAction = this.getBucketlistItemsAction.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.addItemToBucketlist = this.addItemToBucketlist.bind(this);
        this.addItemToBucketlistAction = this.addItemToBucketlistAction.bind(this);
        this.state = { bucketlist: '', showModal: false, newname: '', items: [], showAdd: false, status: '',
            deadline: '', title: '', notificationSystem: null, showItemPanel: false };
    }
    openAdd(event) {
        event.preventDefault();
        this.setState({ showAdd: true });
    }

    openModal(event) {
        event.preventDefault();
        this.setState({ showModal: true });
    }

    closeModal(event) {
        event.preventDefault();
        this.setState({ showModal: false });
        this.setState({ showAdd: false });
    }

    getName(field, event) {
        const newState = {};
        newState[field] = event.target.value;
        this.setState(newState);
    }

    deleteBucketlist(event) {
        event.preventDefault();
        const bID = this.refs.bid.value;
        alert(`Do you really want to delete this item? : ${bID}`);
        const BlData = new FormData();
        BlData.append('bucketlistID', bID);

        this.deleteAction(`${baseUrl}/bucketlists/` + bID.toString(), BlData);
    }
    editBucketlist(event) {
        event.preventDefault();
        const bID = this.refs.bid.value;
        const BlData = new FormData();
        BlData.append('bucketlistID', bID);
        BlData.append('newname', this.state.newname);
        this.updateAction(`${baseUrl}/bucketlists/` + bID.toString(), BlData);
        // this.setState({ newname: '', showModal: false });
    }

    updateAction(url, data) {
        const updateData = {
            method: 'PUT',
            body: data,
            headers: { Authorization: localStorage.getItem('token') },
        };
        return fetch(url, updateData)
            .then(response => response.json()).then(res => {
                this.props.getBucketlists(`${baseUrl}/bucketlists/`);
                if (res.message === 'Bucketlist updated successfully'){
                    this.state.notificationSystem.addNotification({
                        message: res.message,
                        level: 'success'
                    });
                    this.setState({ newname: '', showModal: false });
                } else {
                    this.state.notificationSystem.addNotification({
                        message: res.message,
                        level: 'error'
                    });
                    this.setState({ newname: '' });
                }
            });
    }

    componentDidMount(){
        this.setState({ notificationSystem: this.refs.notificationSystem });
    }
    delete(url, data) {
        return fetch(url, data)
            .then(response => response.json()).then(res => {
                this.props.getBucketlists(`${baseUrl}/bucketlists/`);
                console.log(res)
            });
    }

    deleteAction(url, data) {
        // console.log('URL', url);
        const deleteData = {
            method: 'DELETE',
            body: data,
            headers: { Authorization: localStorage.getItem('token') },
        };
        return this.delete(url, deleteData)
        // return fetch(url, deleteData)
        //     .then(response => response.json()).then(res => {
        //         this.props.getBucketlists(`${baseUrl}/bucketlists/`);
        //         // console.log(res)
        //     });
    }

    getBucketlistItems(event) {
        event.preventDefault();
        const bID = this.refs.bid.value;
        const BlData = new FormData();
        BlData.append('bucketlistID', bID);
        this.getBucketlistItemsAction(`${baseUrl}/bucketlists/`+ bID.toString(), BlData);
        this.setState({ showItemPanel: !this.state.showItemPanel })
    }

    getBucketlistItemsAction(url) {
        const getData = {
            method: 'GET',
            headers: { Authorization: localStorage.getItem('token') },
        };
        return fetch(url, getData)
            .then(response => response.json()).then((res) => {
                    if (!res['message']){
                        this.setState({ items: res });
                    }

                    if (Object.keys(this.state.items).length===0){
                        this.setState({showItemPanel: false});
                        this.state.notificationSystem.addNotification({
                            message: 'Bucketlist is Empty!',
                            level: 'warning'
                        });
                    } // else {
                    // console.log(this.state.items);
                    // }
                },
            );
    }
    handleChange(field, event) {
        const newState = {};
        newState[field] = event.target.value;
        this.setState(newState);
    }

    addItemToBucketlist(event) {
        event.preventDefault();
        const bID = this.refs.bid.value;
        const blItemData = new FormData();
        blItemData.append('title', this.state.title);
        blItemData.append('deadline', this.state.deadline);
        blItemData.append('status', this.state.status);
        this.addItemToBucketlistAction(`${baseUrl}/bucketlists/` + bID.toString()+ '/items/', blItemData);
        this.setState({ showAdd: false, title:'', deadline: '', status:'', showItemPanel: false });
    }

    addItemToBucketlistAction(url, data) {
        const postData = {
            method: 'POST',
            body: data,
            headers: { Authorization: localStorage.getItem('token') },
        };
        return fetch(url, postData)
                 .then(response => response.json()).then((response) => {
            if(response.message === "Bucketlist item added successfully"){
                this.state.notificationSystem.addNotification({
                    message: 'Bucketlist item added successfully!',
                    level: 'success'
                });
            } else {
                this.state.notificationSystem.addNotification({
                    message: response.message,
                    level: 'error'
                });
            }
            })
    }
    render() {
        const addBucketlistItemtooltip = (
            <Tooltip id="tooltip">Add item to this bucketlist</Tooltip>
        );
        const editBucketlisttooltip = (
            <Tooltip id="tooltip">Edit bucketlist</Tooltip>
        );
        const deleteBucketlisttooltip = (
            <Tooltip id="tooltip">Delete bucketlist.</Tooltip>
        );
        return (
            <li className="list-group-item col-md-12 item">
                <a onClick={this.getBucketlistItems}>{this.props.bucketlist}</a>
                <div className="pull-right">
                    <form className="form-inline pull-right" >
                        <input
                            type="hidden"
                            name="name"
                            value={this.props.id}
                            ref="bid"
                            required
                        />
                        <OverlayTrigger placement="top" overlay={addBucketlistItemtooltip}>
                        <Button type="submit" className="btn btn-sm btn-success btn-circle"  id="add" onClick={this.openAdd}>
                            <i className="glyphicon glyphicon-plus" /></Button>
                        </OverlayTrigger>
                        <OverlayTrigger placement="top" overlay={editBucketlisttooltip}>
                        <Button type="submit" className="btn btn-sm btn-primary btn-circle" id="edit" onClick={this.openModal}>
                            <i className="glyphicon glyphicon-edit" /></Button>
                        </OverlayTrigger>
                        <OverlayTrigger placement="top" overlay={deleteBucketlisttooltip}>
                        <Button type="submit" className="btn btn-sm btn-danger btn-circle" id="delete" onClick={this.deleteBucketlist}>
                            <i className="glyphicon glyphicon-trash" /></Button>
                        </OverlayTrigger>
                    </form>
                    <Modal show={this.state.showModal} onHide={this.closeModal} bsSize="small" aria-labelledby="contained-modal-title-sm" >
                        <Modal.Header closeButton>
                            <Modal.Title>Update Bucketlist: {this.props.bucketlist}</Modal.Title>
                        </Modal.Header>
                        <Modal.Body>
                            <form onSubmit={this.handleAdd}>
                                <div className="form-group">
                                    <label htmlFor="name">New Name:</label>
                                    <input
                                        type="text"
                                        className="form-control"
                                        id="name"
                                        defaultValue={this.props.bucketlist}
                                        onChange={this.getName.bind(this, 'newname')}
                                    />
                                </div>
                                <button type="submit" className="btn btn-default" onClick={this.editBucketlist}>Update Bucketlist</button>
                            </form>
                        </Modal.Body>
                    </Modal>
                    <Modal show={this.state.showAdd} onHide={this.closeModal} bsSize="small" aria-labelledby="contained-modal-title-sm">
                        <Modal.Header closeButton>
                            <Modal.Title>Add item to {this.props.bucketlist}</Modal.Title>
                        </Modal.Header>
                        <Modal.Body>
                            <form onSubmit={this.addItemToBucketlist} className="addItemForm">
                                <div className="form-group">
                                    <label htmlFor="title">Title:</label>
                                    <input
                                        type="text"
                                        className="form-control"
                                        id="name"
                                        value={this.state.title}
                                        onChange={this.handleChange.bind(this, 'title')}
                                    />
                                </div>
                                <div className="form-group">
                                    <label htmlFor="Deadline">Deadline:</label>
                                    <input
                                        type="date"
                                        className="form-control"
                                        id="deadline"
                                        value={this.state.deadline}
                                        onChange={this.handleChange.bind(this, 'deadline')}
                                    />
                                </div>
                                <div className="form-group">
                                    <label htmlFor="status">Status:</label>
                                    <select name="status"
                                            onChange={this.handleChange.bind(this, 'status')}
                                            value={this.state.status}>
                                        <option value="">Select option</option>
                                        <option value="Incomplete">Incomplete</option>
                                        <option value="Complete">Complete</option>
                                    </select>
                                </div>
                                <button type="submit" className="btn btn-default">Add item</button>
                            </form>
                        </Modal.Body>
                    </Modal>
                </div>
                <NotificationSystem ref="notificationSystem" />
                <BucketlistItems bID={this.props.id} items={this.state.items}
                                 token={this.props.token}
                                 showItemPanel={this.state.showItemPanel}
                                 getBucketlistItems={this.getBucketlistItems}
                />
            </li>
        );
    }
}