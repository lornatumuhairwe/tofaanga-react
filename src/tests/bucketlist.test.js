import React from 'react';
import renderer from 'react-test-renderer';
import Bucketlist from '../components/Bucketlist/bucketlist';
import { mount } from 'enzyme';
import { stub } from 'sinon';

global.localStorage = {
  setItem: () => {}, getItem: () => {},
};

describe('bucketlist snapshot', () => {
  const isLoggedIn = true;
  const displaySignup = () => {};
  beforeEach(() => {
    stub(Bucketlist.prototype, 'getBucketlists').returns(true);
  });
  it('renders', () => {
    const bucketlist = renderer.create(<Bucketlist isLoggedIn={isLoggedIn} displaySignup={displaySignup} />);
    expect(bucketlist.toJSON()).toMatchSnapshot();
  });

  // it('button opens modal onclick', () => {
  //   const bucketlist = mount(<Bucketlist />);
  // });
});
