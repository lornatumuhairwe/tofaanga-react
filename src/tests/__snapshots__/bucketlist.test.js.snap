// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bucketlist snapshot renders 1`] = `
<div>
  <div
    className="notifications-wrapper"
    style={Object {}}
  />
  <nav
    className="navbar navbar-inverse"
  >
    <ul
      className="nav navbar-nav"
    >
      <li>
        <a
          href=""
        >
          TOFAANGA
        </a>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right"
    >
      <li
        id="logoutBtn"
      >
        <a
          onClick={[Function]}
        >
          <span
            className="glyphicon glyphicon-log-out"
          />
        </a>
      </li>
    </ul>
    <div>
      <form
        className="navbar-form navbar-right"
      >
        <div
          className="form-group"
        >
          <span
            className="input-group"
          >
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              type="text"
            />
            <span
              className="input-group-btn"
            >
              <button
                className="btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <i
                  className="glyphicon glyphicon-search"
                />
              </button>
            </span>
          </span>
        </div>
      </form>
    </div>
  </nav>
  <button
    className="btn btn-sm col-md-offset-1 col-md-1"
    id="addBucketlist"
    onClick={[Function]}
  >
    <i
      className="glyphicon glyphicon-plus-sign"
    />
    Add Bucketlist
  </button>
  <div
    className="loader"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
        <path opacity=\\".25\\" d=\\"M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4\\"/>
        <path d=\\"M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z\\">
          <animateTransform attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 16 16\\" to=\\"360 16 16\\" dur=\\"0.8s\\" repeatCount=\\"indefinite\\" />
        </path>
      </svg>
      ",
      }
    }
    style={
      Object {
        "fill": "red",
        "height": 50,
        "width": 50,
      }
    }
  />
  <div>
    <ul
      className="list-group col-md-12"
      id="bucketlists"
    />
  </div>
</div>
`;
